pkg/machine/ocipull/ociartifact.go

```diff
@@ -92,9 +89,23 @@ func NewOCIArtifactPull(ctx context.Context, dirs *define.MachineDirs, endpoint
 
 	cache := false
 	if endpoint == "" {
-		imageName := artifactImageName
-		endpoint = fmt.Sprintf("docker://%s/%s/%s:%s", artifactRegistry, artifactRepo, imageName, artifactVersion.majorMinor())
-		cache = true
+		return nil, fmt.Errorf("no machine image endpoint provided")
+	}
+
+	// Strip docker:// prefix to parse the reference
+	if strings.HasPrefix(endpoint, "docker://") {
+		ref, err := reference.ParseNormalizedNamed(strings.TrimPrefix(endpoint, "docker://"))
```

Comment:
use strings.CutPrefix() which combines the has prefix and trim prefix into one function so the string is only walked once.

```
if image,ok := strings.CutPrefix(endpoint, "docker://"); ok {
reference.ParseNormalizedNamed(image)
```